api/proc.capnp

```diff
@@ -0,0 +1,47 @@
+using Go = import "/go.capnp";
+
+@0xd78885a0de56b292;
+
+$Go.package("proc");
+$Go.import("github.com/wetware/ww/internal/api/proc");
+
+using Chan = import "channel.capnp";
+
+
+interface Executor(T) {
+    exec @0 (param :T) -> (proc :P);
+}
+
+# P is a basic asynchronous process capability.  
+interface P {
+    wait @0 () -> ();
+}
+
+
+interface Unix extends(Executor(Command)) {
+    struct Command {
+        path @0 :Text;
+        dir  @1 :Text;
+        args @2 :List(Text);
+        env  @3 :List(Text);
+
+        stdin  @4 :StreamReader;
+        stdout @5 :StreamWriter;
+        stderr @6 :StreamWriter;
+    }
+
+    interface Proc extends(P) {
+        signal @0 (signal :Signal);
+        enum Signal {
+            sigINT  @0;
+            sigTERM @1;
+            sigKILL @2;
+        }
+    }
+
+    interface StreamReader {
```

Comment:
Tbh this interface does not convince me. I'm not sue if it's the naming or the extra-complexity, but I think is not enough intuitive

Reply:
Yeah, I'm not vastly fond of it either.  It also makes the implementation of the process more complex, since a call to `Exec` now does two things:  start the process, and return a capability for stdin.

I don't see a simpler way to do it that also preserves the streaming semantics.  Do you have any suggestions?

Alternatively, perhaps the problem is that `StreamReader` is poorly named?  It's not exactly "something that reads from a stream".  Rather, it's something approximating a `WriterCallback`.  Do you think changing the name would help clarify things?